mllib/src/main/scala/org/apache/spark/ml/fpm/AssociationRules.scala

```diff
@@ -0,0 +1,234 @@
+import org.apache.spark.ml.util.{DefaultParamsReader, DefaultParamsWriter, _}
+import org.apache.spark.mllib.fpm.{AssociationRules => MLlibAssociationRules}
+import org.apache.spark.mllib.fpm.FPGrowth.FreqItemset
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+
+
+/**
+ * Common params for AssociationRules and AssociationRulesModel
+ */
+private[fpm] trait AssociationRulesParam extends Params with HasPredictionCol{
+
+  /**
+   * Param for items column name. Items must be array of Strings.
+   * Default: "items"
+   *
+   * @group param
+   */
+  final val itemsCol: Param[String] = new Param[String](this, "itemsCol", "column name in the" +
+    " DataFrame containing the items")
+
+
+  /** @group getParam */
+  @Since("2.2.0")
+  final def getItemsCol: String = $(itemsCol)
+  setDefault(itemsCol -> "items")
+}
+
+@Since("2.2.0")
+object AssociationRules extends DefaultParamsReadable[AssociationRules] {
+
+  @Since("2.2.0")
+  override def load(path: String): AssociationRules = super.load(path)
+}
+
+/**
+ * :: Experimental ::
+ *
+ * Generates association rules from frequent itemsets DataFrame("items", "freq"). This method only
+ * generates association rules which have a single item as the consequent.
+ */
+@Since("2.2.0")
+@Experimental
+class AssociationRules(override val uid: String)
+  extends Estimator[AssociationRulesModel] with AssociationRulesParam with HasPredictionCol {
+
+  @Since("2.2.0")
+  def this() = this(Identifiable.randomUID("AssociationRules"))
+
+  /** @group setParam */
+  @Since("2.2.0")
+  def setItemsCol(value: String): this.type = set(itemsCol, value)
+
+  /**
+   * Param for frequency column name. Data type should be Long.
+   * Default: "freq"
+   *
+   * @group param
+   */
+  final val freqCol: Param[String] = new Param[String](this, "freqCol", "frequency column name")
+
+
+  /** @group getParam */
+  @Since("2.2.0")
+  final def getFreqCol: String = $(freqCol)
+
+  /** @group setParam */
+  @Since("2.2.0")
+  def setFreqCol(value: String): this.type = set(freqCol, value)
+  setDefault(freqCol -> "freq")
+
+  /**
+   * Param for minimum confidence, range [0.0, 1.0].
+    *
+    * @group param
+   */
+  final val minConfidence: DoubleParam = new DoubleParam(this, "minConfidence", "min confidence",
+    ParamValidators.inRange(0.0, 1.0))
+
+  /** @group getParam */
+  @Since("2.2.0")
+  final def getMinConfidence: Double = $(minConfidence)
+
+  /** @group setParam */
+  @Since("2.2.0")
+  def setMinConfidence(value: Double): this.type = set(minConfidence, value)
+  setDefault(minConfidence -> 0.8)
+
+  /**
+   * Computes the association rules with confidence above [[minConfidence]].
+   *
+   * @return a DataFrame("antecedent", "consequent", "confidence") containing the association
+   *         rules.
+   */
+  @Since("2.2.0")
+  override def fit(dataset: Dataset[_]): AssociationRulesModel = {
```

Comment:
Definitely document the input schema since it is non-standard.  The description here can be short, with more details in the class-level Scala doc.